plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -2087,6 +2148,26 @@ private void verifySplitCount(String query, int expectedSplitCount, Materialized
         });
     }
 
+    private void verifyPredicatePushdownDataRead(@Language("SQL") String queryWithPushdown, @Language("SQL") String queryWithoutPushdown, boolean supportsPushdown)
+    {
+        try {
+            ResultWithQueryId<MaterializedResult> resultWithPredicatePushdown = getDistributedQueryRunner().executeWithQueryId(getSession(), queryWithPushdown);
+            ResultWithQueryId<MaterializedResult> resultWithoutPredicatePushdown = getDistributedQueryRunner().executeWithQueryId(getSession(), queryWithoutPushdown);
+
+            DataSize withPushdownDataSize = getOperatorStats(resultWithPredicatePushdown.getQueryId()).getInputDataSize();
+            DataSize withoutPushdownDataSize = getOperatorStats(resultWithoutPredicatePushdown.getQueryId()).getInputDataSize();
```

Comment:
stress test that to verify that if test is flaky the pattern in `@Flaky` annotation is the correct one.